### ISSUE TITLE
Replace all Yes/No/OK/Cancel dialogs

### ISSUE THREAD

**Issue by oscargus** [koppor/jabref#149]
As brought up by @Siedlerchr in https://github.com/JabRef/jabref/pull/1683  http://ux.stackexchange.com/questions/756/what-are-some-alternatives-to-the-phrase-are-you-sure-you-want-to-xyz-in-confi

We currently have 50 instances of `JOptionPane.showConfirmDialog` where for sure this is not followed.

**Comment by Siedlerchr** [comment]
I already documented this in our wiki, too  ;)

**Comment by oscargus** [comment]
Yes, but good to have issues if we should change the code, not just write good new code. ;-)

It is, however, not enough to only search for `showConfirmDialog`:
``` Java
                int answer = JOptionPane.showOptionDialog(KeyBindingsDialog.this,
                        Localization.lang("All key bindings will be reset to their defaults.") + " " +
                        Localization.lang("Continue?"),
                        Localization.lang("Resetting all key bindings"),
                        JOptionPane.YES_NO_OPTION,
                        JOptionPane.QUESTION_MESSAGE, null,
                        new String[] {Localization.lang("OK"),
                                Localization.lang("Cancel")},
                        Localization.lang("OK"));
```

**Comment by Siedlerchr** [comment]
@oscargus  Maybe you could add some sample code for a correct dialog in the wiki section?

**Comment by Siedlerchr** [comment]
Windows Design Guidelines also have a section about this: https://msdn.microsoft.com/de-de/library/windows/desktop/dn742470.aspx

**Comment by koppor** [comment]
(jsut adding links)

- This is also outlined at https://ux.stackexchange.com/a/768/93436
- Written in (very) short at https://github.com/JabRef/jabref/blob/main/docs/advanced-reading/ui-recommendations.md#designing-gui-confirmation-dialogs

**Comment by BugMan-111** [comment]
Hello, I am currently could you please move this issue to reserved? thanks

**Comment by koppor** [comment]
For reference, this is how KeePass does it; which we DO NOT want to do:

![grafik](https://github.com/koppor/jabref/assets/1366654/8a3fc70f-4f17-4304-8935-3f5acca6957f)

**Comment by colinldbd** [comment]
Hi, I'm a new contributor and for a university assignment I would like to learn how to contribute to an open-source project. I'll read through the contributor's guide carefully and give this issue a go. 

Can you please assign this issue to me? I also commented on another first good issue, maybe I'll give them both a try if possible.

**Comment by SUPWEZZ** [comment]
Hi, I would like the work on the issue, please assign it to me if possible!

**Comment by SUPWEZZ** [comment]
Hi @koppor, I would like the work on the issue, please assign it to me if possible!

**Comment by koppor** [comment]
@SUPWEZZ @colinldbd was quicker than you, so I assign @colinldbd

**Comment by colinldbd** [comment]
@Siedlerchr @oscargus Hi, I see that you might have added some sample code for a correct dialog in the wiki section, can I get a link to where that is? 
Many thank!

**Comment by koppor** [comment]
We only have some hints at https://devdocs.jabref.org/code-howtos/ui-recommendations.html. There, we link to https://ux.stackexchange.com/questions/756/what-are-some-alternatives-to-the-phrase-are-you-sure-you-want-to-xyz-in-confi/768#768 as an example to replace "OK/Cancel" by "Post/Cancel"

Note, the hard task is to find the dialogs, understand their context and to rename the buttons. The easy task is then to implement the renamings. -- This issue covers the hard part, too. In case this is too hard, be invited to check other issues.
